src/igvfd/upgrade/sample.py

```diff
@@ -513,3 +513,15 @@ def in_vitro_system_23_24(value, system):
             f' Please make sure it is correct before removing the notes.'
         )
         value['notes'] = notes.strip()
+
+
+@upgrade_step('primary_cell', '21', '22')
+@upgrade_step('in_vitro_system', '24', '25')
+def sample_21_22(value, system):
+    # https://igvf.atlassian.net/browse/IGVF-2310
+    if 'calcified' in value['biosample_qualifiers']:
```

Comment:
'biosample_qualifiers' is not required so need to implement something in case it does not exist in the object.
Like: `if 'calcified' in value.get('biosample_qualifiers', [])`